File: PasswordChecker/Proactive/python/ProactivePasswordChecker/ProactivePasswordChecker/checker.py
from .Analyses import simple_analysis, char_class_analysis, sequence_analysis, substring_analysis, word_use_analysis


def validate_password(password: str):
    # Passwords with length of 2 and lesser are automatically calculated to be 1.0
    if len(password) <= 2:
        return {'result': 1.0}

    # Get result of each analysis
    result = {'simple_analysis': simple_analysis(password),
              'char_class_analysis': char_class_analysis(password),
              'sequence_analysis': sequence_analysis(password),
              'substring_analysis': substring_analysis(password),
              'word_use_analysis': word_use_analysis(password)}

    # Get average result of all the analysis
    result['result'] = float(sum([result[key]['result'] for key in result.keys()]) / 5.0)
    return result

File: PasswordChecker/Proactive/php/skutam/proactive-password-checker/src/checker.php
<?php

namespace ProactivePasswordChecker;

include_once(__DIR__ . '/Analyses/CharClassAnalysis.php');
include_once(__DIR__ . '/Analyses/SequenceAnalysis.php');
include_once(__DIR__ . '/Analyses/SimpleAnalysis.php');
include_once(__DIR__ . '/Analyses/SubstringAnalysis.php');
include_once(__DIR__ . '/Analyses/WordUseAnalysis.php');

function validate_password(string $password) {
    if (mb_strlen($password) <= 2) {
        return array('result' => 1.0);
    }

    $result = array(
        'simple_analysis' => simple_analysis($password),
        'char_class_analysis' => char_class_analysis($password),
        'sequence_analysis' => sequence_analysis($password),
        'substring_analysis' => substring_analysis($password),
        'word_use_analysis' => word_use_analysis($password)
    );

    $result['result'] = (
        $result['simple_analysis']['result'] +
        $result['char_class_analysis']['result'] +
        $result['sequence_analysis']['result'] +
        $result['substring_analysis']['result'] +
        $result['word_use_analysis']['result']
    ) / 5.0;
    return $result;
}

?>